apps/management-app/src/components/Authenticated/CDO/AdsTypesDetail/index.tsx

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { yupResolver } from '@hookform/resolvers/yup';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import { useCallback } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+import * as yup from 'yup';
+import { useAppDispatch } from '@/store';
+import ControlledTextField from '@/components/Common/ControlledTextField';
+import { ModalKey } from '@/constants/modal';
+import {
+  useDeleteAdsTypesMutation,
+  useLazyGetAdsTypeByIdQuery,
+  useUpdateAdsTypeMutation,
+} from '@/store/api/generalManagementApiSlice';
+import { isApiErrorResponse } from '@/store/api/helper';
+import { showModal } from '@/store/slice/modal';
+import { showError } from '@/utils/toast';
+import FormInputSkeleton from '../FormInputSkeleton';
+import StaticActionBar from '../StaticActionBar';
+
+interface FormData {
+  name: string;
+}
+
+const schema = yup.object({
+  name: yup.string().required("Advertisement type's name is required"),
+});
+
+const AdsTypesDetail = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  const [getAdsType, { data }] = useLazyGetAdsTypeByIdQuery();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { isDirty, isValid, isLoading },
+    reset,
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+    mode: 'onChange',
+    defaultValues: async () => {
+      try {
+        const adsType = await getAdsType(parseInt(id!), true).unwrap();
+        return adsType;
+      } catch (error) {
+        showError(
+          isApiErrorResponse(error) && error.status === 404
+            ? 'Detail not found'
+            : 'Something went wrong',
+        );
+        navigate('/ads-types', { replace: true });
+        return { name: '' };
+      }
+    },
+  });
+
+  const [updateAdsType] = useUpdateAdsTypeMutation();
+
+  const handleUpdateAdsType = handleSubmit((data: FormData) => {
+    dispatch(
+      showModal(ModalKey.GENERAL, {
+        headerText: `Apply changes ?`,
+        onModalClose: () => null,
```

Comment:
we have handled close in `src/components/Modals/index.tsx`

Reply:
remove it `onModalClose: () => null,`

Reply:
Updated